osquery/filesystem/tests/filesystem.cpp

```diff
@@ -951,4 +947,57 @@ TEST_F(FilesystemTests, test_directory_listing_with_legacy_logic) {
   deleteDirectoryContent(test_root_work_dirs);
 }
 
+TEST_F(FilesystemTests, test_directory_listing_with_file_symlink) {
+  // This test verifies that a file symlink is not mistaken for a directory.
+  const fs::path test_root_dir = fs::temp_directory_path() / genRandomName();
+  ASSERT_TRUE(fs::create_directory(test_root_dir));
+
+  fs::ofstream test_file(test_root_dir / "test_file.txt");
+  test_file.close();
+
+  // Create symlink
+  try {
+    fs::create_symlink(test_root_dir / "test_file.txt", test_root_dir / "link");
+  } catch (const fs::filesystem_error& e) {
+    FAIL() << "Error creating symlink: " << e.what();
+  }
+
+  std::vector<std::string> found_directories;
+  ASSERT_TRUE(
+      listDirectoriesInDirectory(test_root_dir, found_directories, false));
```

Comment:
Nit: Also test with `recursive=true` (and some link on a subdirectory).

Reply:
Added tests with recursive. Adding a link to subdirectory is done in other tests, so not needed here.